[Bug]: Crashes on  Node v12.13.1

### Expected Behavior

I expect that when running from command line utilizing Nodejs 12.13.1 that the command `depngn 12.13.1` will not crash

### Actual Behavior

It throws an error about not being able to find module 'fs/promises'


### Steps to Reproduce

1. install depngn  globally with npm (`npm i -g depngn`)
2. cd into a repository
3. run the command, `depngn 12.13.1`
4.


### Additional Information

I use a node version manager  or windows and have depngn installed globally with node v12.13.1

### What package manager are you seeing the problem with?

npm

### Relevant log output

```shell
internal/modules/cjs/loader.js:800
    throw err;
    ^

Error: Cannot find module 'fs/promises'
Require stack:
- C:\Users\someuser\bin\nvm\v12.13.1\node_modules\depngn\dist\cli.cjs
- C:\Users\someuser\bin\nvm\v12.13.1\node_modules\depngn\bin\depngn.cjs
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:797:15)
    at Function.Module._load (internal/modules/cjs/loader.js:690:27)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (C:\Users\someuser\bin\nvm\v12.13.1\node_modules\depngn\dist\cli.cjs:8:16)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Function.Module._load (internal/modules/cjs/loader.js:727:14)
    at Module.require (internal/modules/cjs/loader.js:852:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\someuser\\bin\\nvm\\v12.13.1\\node_modules\\depngn\\dist\\cli.cjs',
    'C:\\Users\\someuser\\bin\\nvm\\v12.13.1\\node_modules\\depngn\\bin\\depngn.cjs'
  ]
}
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,7 @@
 # main (unreleased)
 
+-[[bugfix]: support Node versions 10-14](https://github.com/upgradejs/depngn/pull/35)
+
 # 0.3.0
 -[[feature]: add the ability to perform the check in the specified path using --cwd option](https://github.com/upgradejs/depngn/pull/30)
 

--- a/src/cli/reporter/html.ts
+++ b/src/cli/reporter/html.ts
@@ -1,4 +1,4 @@
-import { writeFile } from 'fs/promises';
+import { writeFile } from '../../utils';
 import { CompatData } from '../../types';
 
 export async function createHtml(compatData: Record<string, CompatData>, version: string, path: string = 'compat.html') {

--- a/src/cli/reporter/json.ts
+++ b/src/cli/reporter/json.ts
@@ -1,4 +1,4 @@
-import { writeFile } from 'fs/promises';
+import { writeFile } from '../../utils';
 import { CompatData } from '../../types';
 
 export async function createJson(compatData: Record<string, CompatData>, version: string, path: string = 'compat.json') {
@@ -6,6 +6,7 @@ export async function createJson(compatData: Record<string, CompatData>, version
     node: version,
     dependencies: compatData,
   }, null, 2);
+
   await writeFile(path, out);
   console.log(`File generated at ${path}`);
 }

--- a/src/queries/exec.ts
+++ b/src/queries/exec.ts
@@ -1,4 +1,0 @@
-import { exec } from 'child_process';
-import { promisify } from 'util';
-
-export const asyncExec = promisify(exec);

--- a/src/queries/getDependencies.ts
+++ b/src/queries/getDependencies.ts
@@ -1,4 +1,4 @@
-import { asyncExec } from './exec';
+import { asyncExec } from '../utils';
 import { PackageList, Manager } from '../types';
 
 export async function getDependencies(manager: Manager): Promise<PackageList> {

--- a/src/queries/getEngines.ts
+++ b/src/queries/getEngines.ts
@@ -1,4 +1,4 @@
-import { asyncExec } from './exec';
+import { asyncExec } from '../utils';
 import { Manager, PackageList, PackageManagerName } from '../types';
 
 export async function getEngines(deps: PackageList, manager: Manager) {

--- a/src/queries/getPackageManager.ts
+++ b/src/queries/getPackageManager.ts
@@ -1,4 +1,4 @@
-import { access } from 'fs/promises';
+import { access } from '../utils';
 import { Manager, PackageManagerName } from '../types';
 
 const PACKAGE_MANAGER: Record<string, Manager> = {

--- a/src/queries/index.ts
+++ b/src/queries/index.ts
@@ -1,2 +1,1 @@
 export * from './getCompatData';
-export * from './exec';

--- a/src/utils/index.ts
+++ b/src/utils/index.ts
@@ -1,0 +1,14 @@
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import { writeFile as syncWriteFile, access as syncAccess } from 'fs';
+
+// `fs/promises` is only available from Node v14 onwards
+// so we import the sync versions of `writeFile` and `access` and transform them into 
+// async versions using `promisify` (available from Node v8 onwards)
+// if we ever decided to drop support for Node < v14, we can revert to using `fs/promises`
+//
+// `exec` doesn't have a promise version so we have to promisify it no matter what
+
+export const access = promisify(syncAccess);
+export const asyncExec = promisify(exec);
+export const writeFile = promisify(syncWriteFile);

--- a/src/utils/index.ts
+++ b/src/utils/index.ts
@@ -5,7 +5,7 @@ import { writeFile as syncWriteFile, access as syncAccess } from 'fs';
 // `fs/promises` is only available from Node v14 onwards
 // so we import the sync versions of `writeFile` and `access` and transform them into 
 // async versions using `promisify` (available from Node v8 onwards)
-// if we ever decided to drop support for Node < v14, we can revert to using `fs/promises`
+// if we ever decided to drop support for Node <v14, we can revert to using `fs/promises`
 //
 // `exec` doesn't have a promise version so we have to promisify it no matter what
 

